packages/keycloak-user-management/src/components/forms/UserForm/utils.tsx

```diff
@@ -50,6 +61,52 @@ export const createOrEditPractitioners = async (
   if (!isEditMode) history.push(`${URL_USER_CREDENTIALS}/${payload.identifier}`);
 };
 
+/**
+ * util to help create/edit a keycloak user
+ *
+ * @param keycloakBaseURL -  base url for the keycloak instance
+ * @param keycloakUserPayload - the keycloak user payload
+ * @param isEditMode - whether editing or creating the keycloak user
+ * @param callback - function called when user is successfully posted or updated
+ * @param langObj - the translation store object
+ */
+const createEditKeycloakUser = async (
+  keycloakBaseURL: string,
+  keycloakUserPayload: KeycloakUser,
+  isEditMode: boolean,
+  callback: (id: string) => Promise<void>,
+  langObj = lang
+) => {
+  if (isEditMode) {
+    const serve = new KeycloakService(
+      `${KEYCLOAK_URL_USERS}/${keycloakUserPayload.id}`,
+      keycloakBaseURL
+    );
+    return serve
+      .update(keycloakUserPayload)
+      .then(() => {
+        sendSuccessNotification(langObj.MESSAGE_USER_EDITED);
+        callback(keycloakUserPayload.id).catch(() => sendErrorNotification(langObj.ERROR_OCCURED));
```

Comment:
nitpick but these could be performed in parallel since id is readily available

Reply:
@machariamuguku I nested these calls, such that they will only run once the update/create keycloak user call succeeds.